### DISCUSSION TITLE
XPath for Walt Disney blog thumbnails

### DISCUSSION THREAD

**Discussion by MarcelVersteeg** [FreshRSS/FreshRSS#5686] [Q&A] [answered]
**Describe the bug**
I am scraping the web page [https://www.waltdisney.org/blog](https://www.waltdisney.org/blog) for articles, but using the following settings, the thumbnails are not loaded (I have verified that the XPath does return the correct string):

- XPath for: finding news items (most important) `//article[@class="node-article"]`
- XPath (relative to item) for: item thumbnail `substring(substring-before(substring-after(descendant::div[@class="panel-media cell auto"]/div/div/@style,"url("),");"),2,string-length(substring-before(substring-after(descendant::div[@class="panel-media cell auto"]/div/div/@style,"url("),");"))-2)`

**To Reproduce**
1. Add a new feed using HTML+XPath
2. Enter the mentioned properties (and maybe others) in the indicated fields
3. Submit the settings
4. There are no images shown

**Expected behavior**
The images found via the XPath are shown in the feed list

**Environment information (please complete the following information):**
- Device: PC
- OS: Windows 10
- Browser: Firefox 102ESR
- FreshRSS version: 1.21.0
- Database version: MySQL 5.7.18
- PHP version: PHP 7.4.7
- Installation type: self deployment

**Additional context**
I also see that the `content_bin` field in the `_entry` table is also empty for the articles in this feed, while it contains a BLOB for articles from other feeds.

**Comment by Frenzie** [comment, marked as answer]
Drupal normally uses something like rss.xml for feeds btw. But what do you mean by the correct string exactly? I don't see `data-src` in your XPath there, or is that what you're doing with the whole substring business? I'd think a simple `.//*[@data-src]` would do the trick.

Edit:
PS Drupal uses rss.xml for feeds by default iirc. Maybe there's something useful in that.

**Comment by MarcelVersteeg** [comment]
@Frenzie I do see that the rss.xml is available on the website, but that contains more information than just the blog and I just want the blog.

I was indeed using the complete substring business to get the image from another DOM-element (where it was set as the background image via embedded CSS). The suggested XPath also works and loads the images as expected.

**Comment by Frenzie** [comment]
I'm not really familiar with Drupal. It's possible there are no other feeds if they weren't explicitly enabled. In any case, I'm glad to know I was able to offer a simpler solution in passing.
